File: internal/domain/match.go
package domain

import "time"

type Match struct {
	ID           int64     `json:"id"`
	Player1ID    int64     `json:"player1_id"`
	Player2ID    int64     `json:"player2_id"`
	Player1Score int64     `json:"player1_score"`
	Player2Score int64     `json:"player2_score"`
	WinnerID     *int64    `json:"winner_id,omitempty"`
	PlayedAt     time.Time `json:"played_at"`
}

File: app/domain/match.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class Match(BaseModel):
    id: int | None = None
    player1_id: int
    player2_id: int
    player1_score: int = 0
    player2_score: int = 0
    winner_id: int | None = None
    played_at: datetime = datetime.now()
